src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -2933,6 +2972,555 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
+                            // append op2 as gtUnusedVal
+                            //
+                            // This gives us:
+                            // tmp1 = op1
+                            // unused(op2)
+                            // res  = BinOp(tmp1, op3)
+
+                            spillOp1 = true;
+
+                            std::swap(val1, val2); // 2, 1, 3
+
+                            unusedVal1 = true;
+                            break;
+                        }
+
+                        case TernaryLogicUseFlags::BC:
+                        {
+                            // We're using op2 and op3, so we don't
+                            // need to swap, but we do need to
+                            // append op1 as gtUnusedVal
+                            //
+                            // This gives us:
+                            // * unused(op1)
+                            // * res = BinOp(op2, op3)
+
+                            unusedVal1 = true;
+                            break;
+                        }
+
+                        case TernaryLogicUseFlags::None:
+                        {
+                            // We're not using any operands, so we don't
+                            // need to swap, but we do need push all three
+                            // operands up as gtUnusedVal
+
+                            unusedVal1 = true;
+                            unusedVal2 = true;
+                            unusedVal3 = true;
+                            break;
+                        }
+
+                        default:
+                        {
+                            unreached();
+                        }
+                    }
+
+                    if (spillOp1)
+                    {
+                        impSpillSideEffect(true, verCurrentState.esStackDepth -
+                                                     3 DEBUGARG("Spilling op1 side effects for HWIntrinsic"));
+                    }
+
+                    if (spillOp2)
+                    {
+                        impSpillSideEffect(true, verCurrentState.esStackDepth -
+                                                     2 DEBUGARG("Spilling op2 side effects for HWIntrinsic"));
+                    }
+
+                    op3 = impSIMDPopStack();
+                    op2 = impSIMDPopStack();
+                    op1 = impSIMDPopStack();
+
+                    if (unusedVal1)
+                    {
+                        impAppendTree(gtUnusedValNode(*val1), CHECK_SPILL_NONE, impCurStmtDI);
+                    }
+
+                    if (unusedVal2)
+                    {
+                        impAppendTree(gtUnusedValNode(*val2), CHECK_SPILL_NONE, impCurStmtDI);
+                    }
+
+                    if (unusedVal3)
+                    {
+                        impAppendTree(gtUnusedValNode(*val3), CHECK_SPILL_NONE, impCurStmtDI);
```

Comment:
These `CHECK_SPILL_NONE` don't look right. You need to make sure that everything above the unused operands are checked for interference when you append these `gtUnusedValNode`s.

Reply:
Ah, I see. I copied a case that was already doing `CHECK_SPILL_ALL` in `impSpillSideEffects` which isn't the same as happening here.

Reply:
Fixed.